auth/src/VSCodeAzureSubscriptionProvider.ts

```diff
@@ -237,15 +242,21 @@ export class VSCodeAzureSubscriptionProvider extends vscode.Disposable implement
      *
      * @returns A client, the credential used by the client, and the authentication function
      */
-    private async getSubscriptionClient(tenantId?: string, scopes?: string[]): Promise<{ client: SubscriptionClient, credential: TokenCredential, authentication: AzureAuthentication }> {
+    private async getSubscriptionClient(tenantId?: string, scopes?: string[], session?: vscode.AuthenticationSession, account?: vscode.AuthenticationSessionAccountInformation): Promise<{ client: SubscriptionClient, credential: TokenCredential, authentication: AzureAuthentication }> {
         const armSubs = await import('@azure/arm-resources-subscriptions');
-        const session = await getSessionFromVSCode(scopes, tenantId, { createIfNone: false, silent: true });
         if (!session) {
-            throw new NotSignedInError();
+            if (account) {
+                session = await getSessionFromVSCode(scopes, tenantId, { createIfNone: false, silent: true, account });
+            } else {
+                session = await getSessionFromVSCode(scopes, tenantId, { createIfNone: false, silent: true });
+            }
         }
 
         const credential: TokenCredential = {
             getToken: async () => {
+                if (!session) {
+                    throw new NotSignedInError();
+                }

```

Comment:
We moved this check into this callback. Before we did it right after we fetched the session. I tried it out and both seemed to satisfy TS.

Was there a reason to put it into this callback?

Reply:
When I had it outside the callback the build was failing for me. TS wasn't giving me any errors until I ran `npm run build` then is gave me a session may be undefined within the callback.

Reply:
Ok I'll work on this locally and get back to you